### PR TITLE
Add setup steps for pre-sandbox dependency installation

### PR DESCRIPTION
**Pull request by Gkrumbach07**
## Summary
- Adds a `setup` key to `.agentic-ci/config.yml` that defines commands to run on the host before the agent starts
- Setup steps execute with full network access, enabling dependency installation (e.g. `npm ci`) before sandbox/container isolation
- Works across **all backends**: OpenShell, Podman, and Local
- Supports both GitHub Actions-style objects (`name`/`run`) and bare-string shorthand
- Implementation lives in the `Backend` base class so all backends inherit it

## Config format

```yaml
# .agentic-ci/config.yml
setup:
  - name: Install dependencies
    run: npm ci
  - npm run build  # bare string shorthand
```

## Changes
- New `src/agentic_ci/config.py` — config parser for `.agentic-ci/config.yml`
- `src/agentic_ci/backend.py` — `_run_setup_steps()` in the base class
- `src/agentic_ci/backends/openshell/__init__.py` — calls setup steps before workdir upload
- `src/agentic_ci/backends/podman.py` — calls setup steps before container creation
- `src/agentic_ci/backends/local.py` — calls setup steps before agent launch
- New `docs/configuration.md` — project configuration docs (setup steps + network policy)
- Updated `docs/backends/openshell.md`, `docs/index.md`, `AGENTS.md`
- E2E tests for all three backends
- 13 unit tests covering all config forms and edge cases

## Motivation

Repos with heavy dependency installs (e.g. ODH dashboard needing `npm install`) fail in autofix because the OpenShell sandbox restricts network access. This feature runs setup commands outside the sandbox so their outputs are available to the agent.

Slack thread: https://redhat-internal.slack.com/archives/C0ASJ32PJ0N/p1783524952468199
Resolves: [RHAIFIRST-267](https://redhat.atlassian.net/browse/RHAIFIRST-267)
Supersedes: #241

## Test plan
- [x] 13 unit tests (config parsing, edge cases, malformed YAML, type validation)
- [x] Full test suite passes (740 tests)
- [x] Ruff lint + format clean
- [x] Mypy typecheck clean
- [x] E2E test: openshell — marker file created by setup step, agent verifies inside sandbox
- [x] E2E test: podman — marker file created by setup step, agent verifies inside container
- [x] E2E test: local — marker file created by setup step, agent verifies in workdir

🤖 Generated with [Claude Code](https://claude.com/claude-code)

[RHAIFIRST-267]: https://redhat.atlassian.net/browse/RHAIFIRST-267?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

## Summary by CodeRabbit

- **New Features**
  - Added repository setup steps configured through `.agentic-ci/config.yml`.
  - Supports command-based setup steps in local, Podman, and OpenShell runs.
  - Setup commands run before workdir upload or agent execution, enabling dependency installation and preparation.
  - Added OpenShell project network policy configuration through `.agentic-ci/openshell-policy.yml`.

- **Documentation**
  - Added comprehensive project configuration guidance, examples, and workflow details.
  - Updated feature listings and OpenShell backend documentation.

- **Tests**
  - Added unit and end-to-end coverage for setup steps across supported backends.

### CODE REVIEW

File: tests/e2e/e2e-claude-runner.sh [hunk 1]
```diff
@@ -143,6 +143,35 @@ fi
 # Stop the container before the next test
 agentic-ci stop --harness claude-code 2>/dev/null || true
 
+# -- Setup steps test ---------------------------------------------------------
+print_header "=== agentic-ci run: setup steps (podman) ==="
```
Comment (EmilienM) [inline tests/e2e/e2e-claude-runner.sh:147]: I'm trying to figure out where that was executed in https://github.com/opendatahub-io/agentic-ci/actions/runs/29348957409/job/87140253382?pr=247

Comment (EmilienM) [reply tests/e2e/e2e-claude-runner.sh:147]: ah found it

**Review by EmilienM — APPROVED** [review]
